report/5.Solution/DockerBuild.tex

```diff
@@ -0,0 +1,107 @@
+It also includes an additional nxtOSEK requirement, texinfo, which is the official GNU documentation format\cite{texinfo}.
+
+\begin{lstlisting}[language=docker,label={lst:dockerimplementation2},caption={Installation of Wine and texinfo}]
+
+    # Install wine
+    RUN dpkg --add-architecture i386
+    RUN wget -nc https://dl.winehq.org/wine-builds/Release.key
+    RUN apt-key add Release.key
+    RUN apt-add-repository https://dl.winehq.org/wine-builds/ubuntu/
+    RUN apt-get -y install apt-transport-https
+    RUN apt-get update
+    RUN apt-get -y install --install-recommends winehq-staging
+    
+    # Download a specific version of texinfo, the one in 14.04 
+    # sources is not compatible with the expected version
+    RUN wget http://ftp.gnu.org/gnu/texinfo/texinfo-4.13.tar.gz
+    RUN gzip -dc < texinfo-4.13.tar.gz tar -xf -
+    RUN cd texinfo-4.13 \&\& ./configure \&\& make \&\& make install
+\end{lstlisting} 
+Next, the ARM tool chain is installed.
+The tool chain is used by nxtOSEK to build for the ARM7 CPU in the NXT.
+\begin{lstlisting}[language=docker,label={lst:dockerimplementation3},caption={Building the ARM toolchain}]
+    # Build arm toolchain from nxtOSEK
+    COPY build_arm_toolchain.sh home/
+    RUN chmod 755 home/build_arm_toolchain.sh
+    RUN home/build_arm_toolchain.sh
+    
+    # Add build.sh
+    WORKDIR /home/nxtosek
+    COPY build.sh ./
+    RUN chmod 755 ./build.sh
+\end{lstlisting}
+
+Finally, the nxtOSEK files are moved to the home folder, as cleanup, before the entry point for the container is set to the location in the home folder.
+This means that whenever the Docker image is run, any arguments are passed along to the build file in the home directory.
+
+\begin{lstlisting}[language=docker,label={lst:dockerimplementation4},caption={Moving the nxtOSEK files and setting the entrypoint}]   
+    # Move nxtOSEK files to home folder
+    RUN useradd nxtosek
+    RUN usermod --password $(echo nxtosek openssl passwd -1 -stdin) nxtosek
+    RUN chown -R nxtosek ./
+    USER nxtosek
+    ENV HOME /home/nxtosek
+    ADD nxtOSEK.tar.xz ./
+    ADD ecrobot.mak nxtOSEK/ecrobot
+    
+    # Set build.sh as entrypoint to execute when executing docker container
+    ENTRYPOINT ["/home/nxtosek/build.sh"]
+    CMD []    
+\end{lstlisting}
+It is important to mention that all of the downloaded files are cached, so the setup is not run every single time, and a cached version is used instead.
+
+The Docker container allows the group to easily build the NXT implementation from any machine, as long as Docker is supported by the OS.
+The Docker image is hosted publicly on the Docker Hub at \url{https://hub.docker.com/r/teknight/nxtosek/}.
+
+In addition to local builds, the Docker image is also hosted on the TeamCity server, as part of the continuous integration for the project, meaning that the Docker image is run every time something is pushed to a branch.
```

Comment:
In addition to local builds, the Docker image is also ~hosted~used on the TeamCity server, as part of the continuous integration for the project, meaning that the Docker image is ~run~used to build the nxtOSEK code every time something is pushed to a branch.